internal/batcher/batcher_test.go

```diff
@@ -112,31 +112,28 @@ func TestSaturation(t *testing.T) {
 func TestShutdown(t *testing.T) {
 	ctx := context.Background()
 	var nAdds int64 // atomic
-	b := New(int(0), 10, func(x interface{}) error {
+	c := make(chan int, 10)
+	b := New(int(0), cap(c), func(x interface{}) error {
+		for range x.([]int) {
+			c <- 0
+		}
 		atomic.AddInt64(&nAdds, int64(len(x.([]int))))
-		// Make sure some Adds are active on Shutdown.
-		time.Sleep(100 * time.Millisecond)
+		time.Sleep(time.Second) // we want Adds to be active on Shutdown
```

Comment:
How do we know one second is enough time?

Reply:
I can't be absolutely sure, but that's more than enough time to execute a couple of lines of code.

I couldn't figure out how to make this test time-independent.